google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/MessageDispatcher.java

```diff
@@ -299,6 +302,41 @@ int getMessageDeadlineSeconds() {
     return messageDeadlineSeconds.get();
   }
 
+  @InternalApi
+  void setEnableExactlyOnceDelivery(boolean enableExactlyOnceDelivery) {
+    // Sanity check that we are changing the enableExactlyOnceDelivery state and that we are using
+    // the default min
+    if (enableExactlyOnceDelivery == this.enableExactlyOnceDelivery.get()
+        || !minDurationPerAckExtensionDefaultUsed) {
+      return;
+    }
+
+    // We just need to update the minDurationPerAckExtensionSeconds as the
+    // maxDurationPerAckExtensionSeconds does not change
+    int possibleNewMinAckDeadlineExtensionSeconds;
+
+    if (enableExactlyOnceDelivery) {
+      possibleNewMinAckDeadlineExtensionSeconds =
+          Math.toIntExact(
+              Subscriber.DEFAULT_MIN_ACK_DEADLINE_EXTENSION_EXACTLY_ONCE_DELIVERY.getSeconds());
+    } else {
+      possibleNewMinAckDeadlineExtensionSeconds =
+          Math.toIntExact(Subscriber.DEFAULT_MAX_ACK_DEADLINE_EXTENSION.getSeconds());
```

Comment:
I think this should be set to the DEFAULT_MIN_ACK_DEADLINE_EXTENSION right? Not the max?

Reply:
Yes, my tests did not catch this because the happen to be the same value.